scripts/src/view/renderers.tsx

```diff
@@ -294,11 +295,15 @@ module fugazi.view.renderers {
 			}
 
 			if (fieldsList && fieldsList.length > 0) {
-				this.state.isStruct = true;
-				this.state.fields = collections.map<components.types.Type>();
+				this.setState({
```

Comment:
this looks problematic because `setState` isn't synchronous, and three lines below there's this:
`fieldsList.forEach(map => this.state.fields.extend(map));`